ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/task/GetBlockFromPeerTask.java

```diff
@@ -35,17 +42,23 @@ public class GetBlockFromPeerTask extends AbstractPeerTask<Block> {
   private static final Logger LOG = LoggerFactory.getLogger(GetBlockFromPeerTask.class);
 
   private final ProtocolSchedule protocolSchedule;
+  private final PeerTaskExecutor peerTaskExecutor;
```

Comment:
This feels like it will introduce a cyclic dependency between tasks and executors that will be difficult to unwind later. 

Would it be possible to put the creation of the CompletableFuture elsewhere? This looks like the first time we've wanted to chain tasks together like this and compose them, which is a reasonable functionality to want. Can we push that up a layer, perhaps by allowing a task to return a list of tasks for the upper executor to run instead?

Reply:
Copied from discord discussion:

Ah, I think I understand the confusion. The PeerTaskExecutor is part of the new peer task system added in https://github.com/hyperledger/besu/pull/7628. This aims to replace parts of the old EthTask hierarchy, which is extremely confusingly constructed. The peer communication in GetBlockFromPeerTask will soon also be replaced too.

Once all of the peer communication is replaced, I'd like refactor remaining usages of EthTasks away from the EthTasks hierarchy to a simpler structure